chia/consensus/multiprocess_validation.py

```diff
@@ -221,6 +223,13 @@ async def pre_validate_blocks_multiprocessing(
             block,
             None,
         )
+
+        if block_rec.sub_epoch_summary_included is not None and wp_summaries is not None:
+            idx = int(block.height / constants.SUB_EPOCH_BLOCKS) - 1
+            next_ses = wp_summaries[idx]
+            if not block_rec.sub_epoch_summary_included.get_hash() == next_ses.get_hash():
+                log.error("sub_epoch_summary does not match wp sub_epoch_summary list")
+                return None
```

Comment:
@mariano54 not sure what to do about clean up here
clean the latest batch ? clean all the blocks we received in that sync process ?
they are valid blocks, maybe disconnecting the peer is enough.

Reply:
Not sure what you mean by clean. 
Anyway, I don't think this check belongs here. These are perfectly valid blocks, they are just not corresponding to the weight proof. Can we perform this check after adding the blocks instead? I do think we need to disconnect the peer though, since they lied about the blocks

Reply:
We can perform this check in receive block back instead

Reply:
iirc we said receive_block was not the correct place for this because of the reorgs, i think this shouldn't be in blockchain.py

what i mean by "clean up" -  if the proof of space fails we clean the block records from the block_records dictionary
im wondering if i should do the same

Reply:
@mariano54 Ok i got confused by what we do in line 208, that seems redundant as this is not the blockchain cache we are removing from